dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/info/OpenTelemetryInfoFactoryImpl.java

```diff
@@ -0,0 +1,336 @@
+     * (DisposedOpenTelemetryInfo logs warnings and no-ops when it is used).
+     * The ASL also OpenTelemetryInfo.dispose() which will clean up an open telemetry object.
+     *
+     * To ensure threads do not get the wrong OpenTelemetryInfo from the application metadata, it is stored as an AtomicReference
+     */
+
+    private static final String ENV_DISABLE_PROPERTY = "OTEL_SDK_DISABLED";
+    private static final String CONFIG_DISABLE_PROPERTY = "otel.sdk.disabled";
+    private static final String ENV_METRICS_EXPORTER_PROPERTY = "OTEL_METRICS_EXPORTER";
+    private static final String CONFIG_METRICS_EXPORTER_PROPERTY = "otel.metrics.exporter";
+    private static final String ENV_LOGS_EXPORTER_PROPERTY = "OTEL_LOGS_EXPORTER";
+    private static final String CONFIG_LOGS_EXPORTER_PROPERTY = "otel.logs.exporter";
+    private static final String SERVICE_NAME_PROPERTY = "otel.service.name";
+
+    private static final TraceComponent tc = Tr.register(OpenTelemetryInfoFactoryImpl.class);
+
+    private final MetaDataSlot slotForOpenTelemetryInfoHolder;
+
+    //This contains API calls that change between the upstream open telemetry version.
+    //We get a partially configued SDK Builder from OSGi becase we are in a static context
+    //and do not know which version of mpTelemetry will be in use.
+    @Reference
+    private OpenTelemetrySdkBuilderSupplier openTelemetrySdkBuilderSupplier;
+
+    @Activate
+    public OpenTelemetryInfoFactoryImpl(@Reference MetaDataSlotService slotService) {
+        slotForOpenTelemetryInfoHolder = slotService.reserveMetaDataSlot(ApplicationMetaData.class);
+    }
+
+    @Override
+    public OpenTelemetryInfo getOpenTelemetryInfo() {
+        try {
+            ApplicationMetaData metaData = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData().getModuleMetaData().getApplicationMetaData();
+
+            return getOpenTelemetryInfo(metaData);
+        } catch (Exception e) {
+            Tr.error(tc, Tr.formatMessage(tc, "CWMOT5002.telemetry.error", e));
+            return new ErrorOpenTelemetryInfo();
+        }
+    }
+
+    //A shortcut method to avoid fetching metadata more than we need to.
+    @Override
+    public OpenTelemetryInfo getOpenTelemetryInfo(ApplicationMetaData metaData) {
+        try {
+            OpenTelemetryInfoReference atomicRef = (OpenTelemetryInfoReference) metaData.getMetaData(slotForOpenTelemetryInfoHolder);
+            if (atomicRef == null) {
+                //If this is triggered by internal code that isn't supposed to call ApplicationStateListener.applicationStarting() don't throw an error
+                String j2EEName = metaData.getJ2EEName().toString();
+                if (j2EEName.startsWith("io.openliberty") || j2EEName.startsWith("com.ibm.ws")) {
+                    Tr.info(tc, "CWMOT5100.tracing.is.disabled", j2EEName);
+                    return new DisabledOpenTelemetryInfo();
+                }
```

Comment:
As a follow up, I don't think we want this INFO message emitted for internal applications.